components/org.wso2.carbon.event.outbound.adapter.websubhub/src/main/java/org/wso2/carbon/event/outbound/adapter/websubhub/services/WebSubHubAdapterServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.event.outbound.adapter.websubhub.services;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.event.outbound.adapter.websubhub.WebSubHubAdapterService;
+import org.wso2.carbon.event.outbound.adapter.websubhub.exception.WebSubAdapterClientException;
+import org.wso2.carbon.event.outbound.adapter.websubhub.exception.WebSubAdapterException;
+import org.wso2.carbon.event.outbound.adapter.websubhub.model.EventPayload;
+import org.wso2.carbon.event.outbound.adapter.websubhub.model.SecurityEventTokenPayload;
+
+import java.io.IOException;
+
+import static org.wso2.carbon.event.outbound.adapter.websubhub.util.WebSubHubEventAdapterConstants.ADAPTER_HUB_URL;
+import static org.wso2.carbon.event.outbound.adapter.websubhub.util.WebSubHubEventAdapterConstants.DEREGISTER;
+import static org.wso2.carbon.event.outbound.adapter.websubhub.util.WebSubHubEventAdapterConstants.ErrorMessages.ERROR_DEREGISTERING_HUB_TOPIC;
+import static org.wso2.carbon.event.outbound.adapter.websubhub.util.WebSubHubEventAdapterConstants.ErrorMessages.ERROR_REGISTERING_HUB_TOPIC;
+import static org.wso2.carbon.event.outbound.adapter.websubhub.util.WebSubHubEventAdapterConstants.ErrorMessages.WEB_SUB_BASE_URL_NOT_CONFIGURED;
+import static org.wso2.carbon.event.outbound.adapter.websubhub.util.WebSubHubEventAdapterConstants.REGISTER;
+import static org.wso2.carbon.event.outbound.adapter.websubhub.util.WebSubHubEventAdapterUtil.buildSecurityEventToken;
+import static org.wso2.carbon.event.outbound.adapter.websubhub.util.WebSubHubEventAdapterUtil.handleClientException;
+import static org.wso2.carbon.event.outbound.adapter.websubhub.util.WebSubHubEventAdapterUtil.handleServerException;
+import static org.wso2.carbon.event.outbound.adapter.websubhub.util.WebSubHubEventAdapterUtil.makeAsyncAPICall;
+import static org.wso2.carbon.event.outbound.adapter.websubhub.util.WebSubHubEventAdapterUtil.makeTopicMgtAPICall;
+
+/**
+ * OSGi service for publishing events using web sub hub.
+ */
+public class WebSubHubAdapterServiceImpl implements WebSubHubAdapterService {
+
+    private static final Log log = LogFactory.getLog(WebSubHubAdapterServiceImpl.class);
+    private String webSubHubBaseUrl = null;
+
+    @Override
+    public void publish(EventPayload eventPayload, String topicSuffix, String eventUri) throws WebSubAdapterException {
+
+        //Getting the organization name of Event Payload object since it is the tenant domain.
+        String tenantDomain = eventPayload.getOrganizationName();
+        SecurityEventTokenPayload securityEventTokenPayload =
+                buildSecurityEventToken(eventPayload, eventUri, topicSuffix);
+        makeAsyncAPICall(securityEventTokenPayload, tenantDomain, topicSuffix, getWebSubBaseURL());
```

Comment:
construct the full topic and pass that rather than doing it with the asyc call method

Reply:
https://github.com/wso2-extensions/identity-outbound-adapters/pull/1/commits/3cb452bb085bfec3e826b7c47723c3d161577f0a